builtin/k8s/platform.go

```diff
@@ -367,6 +368,23 @@ func (p *Platform) Deploy(
 
 	timeout := 10 * time.Minute
 
+	cancelMu := &sync.Mutex{}
+	pollingComplete := false
+	go func() {
+		<-ctx.Done()
```

Comment:
@briancain I took a look around Waypoint as you suggested and there are examples of this approach to listening for cancellation in `ui.go` and `server.go`. I think the gist is that it just blocks this goroutine until the signal is sent and the processing continues. I didn't put this in the poll loop because it's possible the user cancels between polls and so execution would not be guaranteed. At least, that's what I observed the way I implemented it. Mabye a PEBKAC problem?